derive/src/lib.rs

```diff
@@ -91,3 +91,27 @@ pub fn pypayload(input: TokenStream) -> TokenStream {
     let input = parse_macro_input!(input);
     derive_impl::pypayload(input).into()
 }
+
+/// use on struct with named fields like `struct A{x:PyRef<B>, y:PyRef<C>}` to impl `Traverse` for datatype.
+///
+/// use `#[notrace]` on fields you wish not to trace
+///
+/// add `trace` attr to `#[pyclass]` to make it impl `MaybeTraverse` that will call `Traverse`'s `traverse` method so make it
+/// traceable(Even from type-erased PyObject)(i.e. write `#[pyclass(trace)]`).
+/// # Example
+/// ```rust, ignore
+/// #[pyclass(module = false, traverse)]
+/// #[derive(Default, PyTraverse)]
+/// pub struct PyList {
+///     elements: PyRwLock<Vec<PyObjectRef>>,
+///     #[notrace]
+///     len: AtomicCell<usize>,
+/// }
+/// ```
+/// This create both `MaybeTraverse` that call `Traverse`'s `traverse` method and `PyTraverse` that impl `Traverse`
+/// for `PyList` which call elements' `traverse` method and ignore `len` field.
+#[proc_macro_derive(PyTraverse, attributes(notrace))]
```

Comment:
```suggestion
#[proc_macro_derive(Traverse, attributes(pytraverse))]
```
we don't do `PyFromArgs` but `FromArgs`. `Traverse` will be enough.